### PR TITLE
Ledger/Speculos TCP APDU transport

### PR DESCRIPTION
**Pull request by e-asphyx**

### CODE REVIEW

File: pkg/vault/ledger/scan.go [hunk 1]
```diff
@@ -103,6 +104,7 @@ func (s *scanner) scan() ([]*deviceInfo, error) {
 	for _, d := range devs {
 		app, dev, err := s.openPath(d.Path)
 		if err != nil {
+			log.Warnf("%s: %v", d.Path, err)
```
Comment (stephengaudet) [inline pkg/vault/ledger/scan.go:107]: thank you!

File: pkg/vault/ledger/ledger/devices.go [hunk 2]
```diff
@@ -20,43 +20,55 @@ type LedgerDeviceInfo struct {
...
-	&LedgerDeviceInfo{
+var (
+	ledgerBlue = LedgerDeviceInfo{
 		ID:                 "blue",
 		ProductName:        "Ledger Blue",
 		ProductIDMM:        0x00,
 		LegacyUSBProductID: 0x0000,
 		USBOnly:            true,
 		MemorySize:         480 * 1024,
-		BlockSize:          4 * 1024,
```
Comment (stephengaudet) [inline pkg/vault/ledger/ledger/devices.go:35]: was removing  BlockSize from the devices' infos an intentional change?  seems unrelated to the PR summary of adding tcp apdu transport

Comment (stephengaudet) [reply pkg/vault/ledger/ledger/devices.go:35]: was sanity tested with Nano S, so, probably fine.

**Review by stephengaudet — APPROVED** [review]
on a high level, the documentation is accurate and the behaviour is correct. thank you for the speculos backend support @e-asphyx ! 

just wanted to double check that the removal of BlockSize from the devices' infos was an intentional change and not by mistake, seems unrelated to the PR.  tested on Ledger X, S, and SP.  

my first look at the code base, so, please take my comments with a grain of salt. passing around the ctx context as a parameter maybe a classic situation for dependency injection.

lastly, I just noticed, there is a failing check, code climate. max number of returns exceeded

**Comment by stephengaudet** [comment]
I just noticed, there is a failing check, code climate. max number of returns exceeded

File: pkg/vault/ledger/ledger/devices.go (hunk 2 above)
Comment (stephengaudet) [reply pkg/vault/ledger/ledger/devices.go:35]: I think it would be better to only make changes required to add speculos support, that way we won't be introducing untested changes.

Comment (stephengaudet) [reply pkg/vault/ledger/ledger/devices.go:35]: this has been tested on nanoS, nanoSP, and nanoX

**Review by stephengaudet — APPROVED** [review]

**Review by jevonearth — APPROVED** [review]
